### DISCUSSION TITLE
Error while building the container during step 7 of the ReadMe Usage section

### DISCUSSION THREAD

**Discussion by WAlbers99** [tudelft/AE4353-Y24#7] [Q&A] [answered]
When building the container, I get the following error:

Error: Command failed: docker buildx build --load --build-arg BUILDKIT_INLINE_CACHE=1 -f /tmp/devcontainercli-wessel/container-features/0.69.0-1726061727646/Dockerfile-with-features -t vsc-ae4353_directory-13808546dd37deb18f873afd5169503081841acfc4c0c2d2d3e2d8e47e431263 --target dev_containers_target_stage --build-context dev_containers_feature_content_source=/tmp/devcontainercli-wessel/container-features/0.69.0-1726061727646 --build-arg _DEV_CONTAINERS_BASE_IMAGE=dev_container_auto_added_stage_label --build-arg _DEV_CONTAINERS_IMAGE_USER=root --build-arg _DEV_CONTAINERS_FEATURE_CONTENT_SOURCE=dev_container_feature_content_temp /home/wessel/AE4353_directory
[2024-09-11T13:38:06.404Z] at FtA (/home/wessel/.vscode-remote-containers/dist/dev-containers-cli-0.384.0/dist/spec-node/devContainersSpecCLI.js:466:1933)

I asked chatgpt but that did not help to solve the issue. What am I doing wrong?

**Comment by Huizerd** [comment]
Hmm, the error is not very informative (it doesn't really say what exactly is going wrong). Could you try rebuilding the container? (Ctrl+Shift+P -> Dev Containers: Rebuild and Reopen in Container)

**Reply by WAlbers99** [reply]
I have done that but I still get the same error. I looked for all the errors in de log file and I found two 
 other errors as well:
1.  ERROR: failed to solve: process "/bin/sh -c apt-get update && export DEBIAN_FRONTEND=noninteractive     && apt-get install -y --no-install-recommends     bzip2     ca-certificates     libglib2.0-0     libsm6     libxext6     libxrender1     mercurial     openssh-client     procps     subversion     tmux     wget     && apt-get upgrade -y     && ln -s /opt/conda/etc/profile.d/conda.sh /etc/profile.d/conda.sh     && apt-get clean -y && rm -rf /var/lib/apt/lists/*" did not complete successfully: exit code: 100
[2024-09-11T13:59:11.687Z] Stop (154124 ms): Run: docker buildx build --load --build-arg BUILDKIT_INLINE_CACHE=1 -f /tmp/devcontainercli-wessel/container-features/0.69.0-1726062988634/Dockerfile-with-features -t vsc-ae4353_directory-13808546dd37deb18f873afd5169503081841acfc4c0c2d2d3e2d8e47e431263 --target dev_containers_target_stage --build-context dev_containers_feature_content_source=/tmp/devcontainercli-wessel/container-features/0.69.0-1726062988634 --build-arg _DEV_CONTAINERS_BASE_IMAGE=dev_container_auto_added_stage_label --build-arg _DEV_CONTAINERS_IMAGE_USER=root --build-arg _DEV_CONTAINERS_FEATURE_CONTENT_SOURCE=dev_container_feature_content_temp /home/wessel/AE4353_directory

2.  => ERROR [dev_container_auto_added_stage_label 5/7] RUN apt-get update  150.2s

**Reply by kmalkow** [reply]
I think it is due to a time mismatch, try this:

**Cause of the Error:**
This error message indicates that the release files for the Ubuntu (WSL2) package repositories are considered "not valid yet" because of a time mismatch between your system clock and the repository servers. This can happen if:

- Your system time is incorrect or out of sync with the real-world time.
- The time zone settings on your system are misconfigured.
- There is a temporary issue with the repository server's release files.

**Steps to Fix the Error:**
1. Check and Sync Your System Time:
Please make sure that your system clock is correctly set and synchronized with the internet time servers. In WSL2, you can manually sync the time using the following commands: 
`sudo apt-get install ntpdate`
`sudo ntpdate time.windows.com`

2. Retry the Update Command Manually:
After syncing the time, try rerunning the update command: `sudo apt-get update && sudo apt-get install`

3. Restart Visual Studio Code:
Restart Visual Studio Code by navigating to the repository directory and executing `code .`

**Comment by WAlbers99** [comment, marked as answer]
In the end, I deleted everything and started from scratch. After that, it worked

**Reply by kmalkow** [reply]
Amazing, glad to hear it!

**Reply by kmalkow** [reply]
We found out the error in the meantime and will update the README so it doesn’t hopefully happen again. Basically, if when you first open the terminal, if you open the WSL terminal and do all the install commands in there, it does everything in root, whereas if you open the Ubuntu terminal (which is the correct one) by typing Ubuntu in your windows search bar and install everything in there, it creates a home/user folder instead of doing everything in root, which is the correct way of doing things. So basically if you do everything in the WSL terminal, the devcontainers image user is in root (_DEV_CONTAINERS_IMAGE_USER=root in the error you posted), whereas is it should be your home/user/…

So basically, if you open the Ubuntu terminal (not the WSL terminal!!!) and install everything in there, everything should work good.
